node/network/statement-distribution/src/lib.rs

```diff
@@ -476,6 +476,21 @@ fn check_statement_signature(
 		.and_then(|v| statement.check_signature(&signing_context, v))
 }
 
+/// Informs all registered listeners about a newly received statement.
+///
+/// Removes all closed listeners.
+async fn inform_statement_listeners(
+	statement: SignedFullStatement,
+	listeners: &mut Vec<mpsc::Sender<SignedFullStatement>>,
+) {
+	// Ignore the errors since these will be removed later.
+	for listener in listeners.iter_mut() {
+		let _ = listener.send(statement.clone()).await;
```

Comment:
a DQ: shouldn't we fan out sending and await the joined future?

Reply:
Something like `futures::stream::iter(listeners.iter_mut()).for_each_concurrent(None, |listener| listener.send(statement.clone())` is probably a good idea. Then again, I have the impression that we're rarely going to have more than a few listeners at a time, so it may not make any practical difference.